pkg/pusher/pusher.go

```diff
@@ -79,6 +85,10 @@ LOOP:
 				break
 			}
 
+			if span == nil {
+				span, _, ctx = s.tracer.StartSpanFromContext(cctx, "pusher-sync-batch", s.logger)
```

Comment:
where will this new ctx be canceled?

Reply:
`ctx` will be on the first run the actual root context (`cctx`) which is being cancelled on `L64`, and once we reach this line then `ctx` becomes a child context of `cctx` and does not have its own cancel function, therefore it is not cancelled. if `cctx` is cancelled, then this context will be cancelled too. at least that is to the best of my understanding

Reply:
SGTM